include/eve/module/core/regular/bit_ternary.hpp

```diff
@@ -0,0 +1,389 @@
+//!       The consteval function mfb (make from bits) can be use to generate this constant from its bits (see below).
+//!     * `x`, `y`, `z`, : first [argument](@ref eve::value).
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
+//!
+//!    **Return value**
+//!
+//!      1. Bitwise ternary logic that provides the capability to implement any three-operand binary function;
+//!        the specific binary function is specified by value of k.
+//!        For each bit in each integer, the corresponding bit from a, b, and c are used according
+//!        to ik, and the result is output.
+//!      2. equivalent to the call on the elements of the tuple.
+//!      3. [The operation is performed conditionnaly](@ref conditional).
+//!
+//!      This is the pattern of a truth table:
+//!
+//!       |  x  |  y  |  z  | result |
+//!       | :-: | :-: | :-: | :----: |
+//!       |  0  |  0  |  0  |    a   |
+//!       |  0  |  0  |  1  |    b   |
+//!       |  0  |  1  |  0  |    c   |
+//!       |  0  |  1  |  1  |    d   |
+//!       |  1  |  0  |  0  |    e   |
+//!       |  1  |  0  |  1  |    f   |
+//!       |  1  |  1  |  0  |    g   |
+//!       |  1  |  1  |  1  |    h   |
+//!
+//!  A programmer as only to supply the result column, i.e. defines values of bits a through h, this is a single 8-bit value
+//!
+//!  `     k = .(a << 7) + (b << 6) + (c << 5) + (d << 4) + (e << 3) + (f << 2) + (g << 1) + h`
+//!
+//!  (a consteval function can be used to do that operation: see the example)
+//!
+//!  @groupheader{Example}
+//!  @godbolt{doc/core/bit_ternary.cpp}
+//================================================================================================
+  inline constexpr auto bit_ternary = functor<bit_ternary_t>;
+//================================================================================================
+//! @}
+//================================================================================================
+
+  namespace detail
+  {
+
+    // this is adapted from Samuel neves ternary logic for sse avx etc.
+    template < int K, typename T0, typename T1, typename T2, callable_options O>
+    EVE_FORCEINLINE  /*bit_value_t<T0, T1, T2>*/ auto bit_ternary_(EVE_REQUIRES(cpu_)
+                                                                  , O const &
+                                                                  , std::integral_constant<int, K> const &
+                                                                  , [[maybe_unused]] T0 const & aa
+                                                                  , [[maybe_unused]] T1 const & bb
+                                                                  , [[maybe_unused]] T2 const & cc
+                                                                  ) noexcept
+    //    requires(K <= 255)
```

Comment:
remove allt he comment here and at line 113

Reply:
done